Instructions in `CONTRIBUTING.md` fail when openssl devel package is not installed

### Component(s)

config/configtls

### Describe the issue you're reporting

Following the instructions in [`CONTRIBUTING.md`](https://github.com/open-telemetry/opentelemetry-collector/blob/346b0657a8e837bbf82f009bde9838a3712ae141/CONTRIBUTING.md) until the recommended [`make`](https://github.com/open-telemetry/opentelemetry-collector/blob/346b0657a8e837bbf82f009bde9838a3712ae141/CONTRIBUTING.md?plain=1#L253) call fails, due to missing openssl dependencies for `config/configtls`.

```
user@pc:~/workspace/otel/opentelemetry-collector$ make -C config/configtls lint
make: Entering directory '/home/user/workspace/otel/opentelemetry-collector/config/configtls'
go tool -modfile /home/user/workspace/otel/opentelemetry-collector/internal/tools/go.mod golangci-lint run
config/configtls/tpm_test.go:28:2: could not import github.com/google/go-tpm/tpm2/transport/simulator (/home/user/go/pkg/mod/github.com/google/go-tpm@v0.9.8/tpm2/transport/simulator/simulator.go:7:2: could not import github.com/google/go-tpm-tools/simulator (/home/user/go/pkg/mod/github.com/google/go-tpm-tools@v0.4.7/simulator/simulator.go:27:2: could not import github.com/google/go-tpm-tools/simulator/internal (-: # github.com/google/go-tpm-tools/simulator/internal
In file included from /home/user/go/pkg/mod/github.com/google/go-tpm-tools@v0.4.7/simulator/ms-tpm-20-ref/TPMCmd/tpm/include/LibSupport.h:65,
                 from /home/user/go/pkg/mod/github.com/google/go-tpm-tools@v0.4.7/simulator/ms-tpm-20-ref/TPMCmd/tpm/include/Tpm.h:47,
                 from /home/user/go/pkg/mod/github.com/google/go-tpm-tools@v0.4.7/simulator/internal/internal_cgo.go:47:
/home/user/go/pkg/mod/github.com/google/go-tpm-tools@v0.4.7/simulator/ms-tpm-20-ref/TPMCmd/tpm/include/Ossl/TpmToOsslSym.h:47:10: fatal error: openssl/aes.h: No such file or directory
   47 | #include <openssl/aes.h>
      |          ^~~~~~~~~~~~~~~
compilation terminated.))) (typecheck)
        "github.com/google/go-tpm/tpm2/transport/simulator"
        ^
make: *** [../../Makefile.Common:71: lint] Error 1
```

Installing OpenSSL headers should be mentioned as a potential dependency to run `make` successfully, as it runs for all moduless, including `config/configtls` by default.

### Tip

<sub>[React](https://github.blog/news-insights/product-news/add-reactions-to-pull-requests-issues-and-comments/) with 👍 to help prioritize this issue. Please use comments to provide useful context, avoiding `+1` or `me too`, to help us triage it. Learn more [here](https://opentelemetry.io/community/end-user/issue-participation/).</sub>

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -231,6 +231,7 @@ Working with the project sources requires the following tools:
 2. [Go](https://go.dev/) (version 1.25 and up)
 3. [GNU Make](https://www.gnu.org/software/make/)
 4. [Docker](https://www.docker.com/)
+5. [OpenSSL Library](https://openssl-library.org/) (for the  `config/configtls` component)
 
 ## Repository Setup
 

